_posts/2018-01-01-futures-0.3.0-alpha.2.md

```diff
@@ -0,0 +1,60 @@
+---
+layout: post
+title:  "Futures 0.3.0-alpha.2"
+author: "Josef Brandl"
+author_github: "MajorBreakfast"
+date:   2018-01-01
+categories: blog
+---
+
+`alpha.2` requires a recent nightly (2018-xx-xx or newer):
+```sh
+$ rustup update
+```
+
+## What's new?
+
+### Refined definition for "task"
+
+The documentation for the `Executor` trait was updated. It now defines a "task executor" like this:
+
+> Futures are polled until completion by tasks, a kind of lightweight "thread". A *task executor* is responsible for the creation of these tasks and the coordination of their execution on real operating system threads. In particular, whenever a task signals that it can make further progress via a wake-up notification, it is the responsibility of the task executor to put the task into a queue to continue executing it, i.e. polling the future in it, later.
+
+The notable difference to before is that the term "task" no longer refers to the future that the task runs. Instead, from now on, the term "task" only refers to the concept of a "lightweight" thread, a kind of thread that doesn't directly correspond to a thread provided by the operating system. This concept is sometimes also referred to as "green thread" - We, however, do not use either these terms because they tend to cause confusion with real operating system threads. Instead, we call this concept a "task" and the thing that runs them "task executor" or often just "executor".
+
+### `Try` impl for `Poll`
+
+The [`Poll`](https://doc.rust-lang.org/nightly/core/task/enum.Poll.html) enum now implements the [`Try`](https://doc.rust-lang.org/std/ops/trait.Try.html) trait. This means you can now do this:
+
+```rust
+fn poll(mut self: PinMut<Self>, cx: &mut task::Context, ) -> Poll<Result<T, E>> {
+    let poll: Poll<Result<T, E>> = self.future().poll();
+    let poll: Poll<T> = poll?; // Returns from function if its an error
+    let ok_value: T = ready!(poll); // Returns from function if not ready
+
+    // Or short:
+    let ok_value = ready!(self.future().poll()?);
+}
+```
+
+- The `?`-operator used on a `Poll<Result<T, E>>` value:
+  - yields `Poll::Pending` from the expression if the value is not ready.
```

Comment:
"yields" is confusing here-- maybe something like

"When `?` is used on a `Poll<Result<T, E>>` or a `Poll<Option<Result<T, E>>>` containing an `Err` value, it will return early. If the expression does not contain an error, it will result in a `Poll<T>` or `Poll<Option<T>>`, respectively."